File: C/2200.c
#include <stdio.h>


 // Definition for a binary tree node.
 struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};



int maxDepth(struct TreeNode* root){
    if(root == NULL)
        return 0;

    int m = maxDepth(root->left);
    int n = maxDepth(root->right);

    if(m >= n)
        return m+1;
    else 
        return n+1;
}

File: GO/2200.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func maxDepth(root *TreeNode) int {
    if root == NULL {
        return 0
    }


    m := maxDepth(root->Left)
    n := maxDepth(root->Right)

    if m>=n {
        return m+1;
    }else {
        return n+1;
    }
}